app/services/arcgis_api/geocoder.rb

```diff
@@ -94,6 +98,8 @@ def faraday
 
         # Parse JSON responses
         conn.response :json, content_type: 'application/json'
+
+        @custom_connection_options.call(conn)
```

Comment:
based on comment about not adding default `|| proc {}` we can make this conditional:

```suggestion
        @custom_connection_options&.call(conn)
```